Tasks/JavaToolInstallerV0/javatoolinstaller.ts

```diff
@@ -55,19 +47,78 @@ async function getJava(versionSpec: string) {
+            .arg('--noprofile')
+            .arg(`--norc`)
+            .arg(escapedScript)
+            .arg(`${versionSpec}`);
+                
+            let result: tr.IExecSyncResult = scriptRunner.execSync();
+            if (result.code != 0) {
+                throw taskLib.loc("AptGetFailed", result.error ? result.error.message : result.stderr);
+            }
+
+            setEnvironmentVars(versionSpec, '/usr/lib/jvm/java-' + versionSpec + '-openjdk-amd64');
+        } else {
+            console.log(taskLib.loc('NotLinuxPlatform'));
+            console.log(taskLib.loc('CurrentPlatformInfo', os.platform()));
+        }
+    } else { //Install JDK via chocolatey command
+        //check that it is Windows
+        if(os.platform() === 'win32') {
+            let filePath = path.join(__dirname, 'chocoJavaInstall.ps1').replace("'", "''");
+            let command = `. '${filePath}' -VersionSpec '${versionSpec}'`;
+            let powershell = taskLib.tool(taskLib.which('pwsh') || taskLib.which('powershell') || taskLib.which('pwsh', true))
+                .arg('-NoLogo')
+                .arg('-NoProfile')
+                .arg('-NonInteractive')
+                .arg('-Command')
+                .arg(command);
+            const options: tr.IExecOptions = <any>{
+                failOnStdErr: false,
+                errStream: process.stdout, // Direct all output to STDOUT, otherwise the output may appear out
+                outStream: process.stdout, // of order since Node buffers it's own STDOUT but not STDERR.
+                ignoreReturnCode: true
+            };
+
+            let exitCode: number = await powershell.exec(options);
+
+            // Fail on exit code.
+            if (exitCode !== 0) {
+                taskLib.setResult(taskLib.TaskResult.Failed, taskLib.loc('JS_ExitCode', exitCode));
+                console.log(taskLib.loc('CurrentPlatformInfo', os.platform()));
+            }
+        } else {
+            console.log(taskLib.loc('NotWindowsPlatform'));
+        }
     }
+}
 
-    let extendedJavaHome = 'JAVA_HOME_' + versionSpec + '_' + taskLib.getInput('jdkArchitectureOption', true);
-    console.log(taskLib.loc('SetJavaHome', jdkDirectory));
-    console.log(taskLib.loc('SetExtendedJavaHome', extendedJavaHome, jdkDirectory));
-    taskLib.setVariable('JAVA_HOME', jdkDirectory);
-    taskLib.setVariable(extendedJavaHome, jdkDirectory);
-    toolLib.prependPath(path.join(jdkDirectory, 'bin'));    
+function cleanDestinationDirectory(cleanDestDir: boolean, extractLocation: string) {
+    // Clean the destination folder before downloading and extracting?
+    if (cleanDestDir && taskLib.exist(extractLocation) && taskLib.stats(extractLocation).isDirectory) {
+        console.log(taskLib.loc('CleanDestDir', extractLocation));
+
+        // delete the contents of the destination directory but leave the directory in place
+        fs.readdirSync(extractLocation)
+            .forEach((item: string) => {
```

Comment:
I would consider `.forEach` deprecated in favor of `for (const item of fs.readdirSync(extractLocation))`